src/world/Units/Players/Player.Legacy.cpp

```diff
@@ -3594,7 +3594,7 @@ void Player::LoadFromDBProc(QueryResultVector & results)
         la.dur = atol(auraDuration.c_str());
 
         std::getline(savedPlayerBuffsStream, auraPositivValue, ',');
-        la.positive = (auraPositivValue.c_str());
+        la.positive = auraPositivValue == "0" ? false : true;
```

Comment:
This and line 4394 should be compared to nullptr, not the string "0".

Reply:
@Evairfairy Are you sure? I inspected this auraPositivValue and it is saved as string and it is always either 0 or 1.

Reply:
@Appled Not without digging into it, but if la.positive is a boolean then the existing check can be rewritten as:

    if (auraPositiveValue.c_str() == nullptr)
        la.positive = false;
    else
        la.positive = true;

So if you're trying to mimic the existing behaviour it should be compared to nullptr.

Reply:
@Evairfairy Sounds reasonable. I'll update it later tonight.

Reply:
@Evairfairy I have amended the change to your solution.